benchmarks/domains/aryl_halides/regression_tl.py

```diff
@@ -0,0 +1,130 @@
+"""TL regression benchmarks for aryl halide reactions."""
+
+from __future__ import annotations
+
+from functools import partial
+
+import pandas as pd
+
+from benchmarks.definition import (
+    TransferLearningRegressionBenchmark,
+    TransferLearningRegressionBenchmarkSettings,
+)
+from benchmarks.definition.base import RunMode
+from benchmarks.definition.regression.core import run_tl_regression_benchmark
+from benchmarks.domains.aryl_halides.core import (
+    load_data,
+    make_objective,
+    make_searchspace,
+)
+
+
+def _aryl_halide_tl_regr(
+    settings: TransferLearningRegressionBenchmarkSettings,
+    source_tasks: list[str],
+    target_tasks: list[str],
+) -> pd.DataFrame:
+    """General benchmark function for aryl halide transfer learning regression.
+
+    This benchmark compares regression performance of TL and non-TL GP models using
+    different aryl halide substrates as tasks. It trains models with varying amounts
+    of source and target data, and evaluates their predictive performance on held-out
+    target data.
+
+    Three different task combinations:
+    - CT_I_BM: Source tasks ["1-chloro-4-(trifluoromethyl)benzene", "2-iodopyridine"]
+            → Target task ["1-iodo-4-methoxybenzene"]
+    - CT_IM: Source task ["1-chloro-4-(trifluoromethyl)benzene"]
+            → Target task ["1-iodo-4-methoxybenzene"]
+    - IP_CP: Source task ["2-iodopyridine"]
+            → Target task ["3-chloropyridine"]
+
+    Key characteristics:
+    • Parameters:
+      - Base: Substance with MORDRED encoding
+      - Ligand: Substance with MORDRED encoding
+      - Additive: Substance with MORDRED encoding
+      - aryl_halide: Task parameter
+    • Target: Reaction yield (continuous)
+    • Objective: Maximization
+    • Compares TL models (SourcePrior, MHGP, SHGP, Index Kernel) vs vanilla GP
+
+    Args:
+        settings: The benchmark settings.
+        source_tasks: List of source task names (aryl halide substrates).
+        target_tasks: List of target task names (aryl halide substrates).
```

Comment:
It's not public package code, so it's not critical, but I want to point out nevertheless as a general comment:
* Try to avoid hard-coding container types for callable arguments – use generic types instead, depending on the requirements. For example, if you hardcode `list[str]`, people will get a linter error if they pass a `tuple[str, ...]` instead. So here, since your consuming function accepts a `Sequence[str]`, use that instead (change suggested above)
* Avoid referencing the container types in the docstring. For example, instead of "List of ...", the following is equally fine but has the advantage that changed in the signature don't require changes in the doctrings (which will always be forgotten)
```suggestion
        source_tasks: Source task names (aryl halide substrates).
        target_tasks: Target task names (aryl halide substrates).
```